test/util/test_runners/integrations_tests_runner.py

```diff
@@ -0,0 +1,138 @@
+        1) Create a netrc file to provide login details
+        2) Run the CLI to initialize the CLI to get the certificate and conjurrc files
+        """
+
+        # copy netrc from home folder to test folder
+        path_provider = TestRunnerPathProvider.getInstance()
+        self.__create_netrc_file()
+        shutil.copy(path_provider.netrc_path,
+                    os.path.join(path_provider.helpers_files_path, path_provider.netrc_file_name))
+
+        proc = sb.Popen(
+            [self.runner_args.working_cli_path_only_for_init, "init", "-a", f"{self.runner_args.account}", "-u",
+             f"{self.runner_args.hostname}",
+             "--force"],
+            stdin=sb.PIPE,
+            stdout=sb.PIPE,
+            stderr=sb.PIPE)
+        output = proc.communicate(input=b'yes\n',timeout=10)[0]
+        if proc.returncode != 0:
+            raise RuntimeError(f"Error Init: {output}")
+        # copy conjurrc from home folder to test folder
+        shutil.copy(path_provider.conjurrc_path, path_provider.test_conjurrc_file_path)
+
+    def __create_netrc_file(self):
+        password = self.__get_api_key()
+        home_path = str(Path.home())
+
+        with open(os.path.join(home_path, '.netrc'), 'w') as f:
+            f.write(f"machine {self.runner_args.hostname}/authn\n")
+            f.write(f"login {self.runner_args.username}\n")
+            f.write(f"password {password}\n")
+
+    def __get_api_key(self):
+        url = f"{self.runner_args.hostname}/authn/cucumber/login"
+        encode = base64.b64encode(f"admin:{self.runner_args.password}".encode('utf-8')).decode('utf-8')
+        headers = {'Authorization': f'Basic {encode}'}
+
+        response = requests.request("GET", url, headers=headers, data={},
+                                    verify=False)  # verify=False == like --insecure
+
+        return response.text
+
+    def __run_test_cases(self, test_cases_class: IntegrationTestCaseBase):
+        """
+        Run class tests.
+         @type test_cases_class: IntegrationTestCaseBase
+        """
+        test_names = self.__get_relevant_tests_for_class(test_cases_class)  # we filter by test_name_identifier
+        suite = unittest.TestSuite()
+        for test_name in test_names:
+            t = test_cases_class(test_name, self.client_params, self.env_params)
+            suite.addTest(t)
+        result = unittest.TextTestRunner().run(suite)
+        return result.wasSuccessful()
+
+    def __get_relevant_tests_for_class(self, class_name: IntegrationTestCaseBase) -> list:
+        """
+        filter out only tests with property named like test_name_identifier
+        this copy the nose2 filtering strategy
+        @param class_name: the class to take the tests from
+        @return: list of all relevant tests
+        """
+        import inspect
+        ret = []
+        all_functions = inspect.getmembers(class_name, predicate=inspect.isfunction)
```

Comment:
should all_functions be all_test_definitions? In other words does function = each individual test to run? If so, let's update to all_test_definitions

and f should be test_definition?

Reply:
yep